### PR TITLE
Acceptance tests for host-specific data loading

### PR DESCRIPTION
**Pull request by vpodzime**
TODO:
- [x] CMDB data is loaded by cf-promises and cf-agent
- [x] vars, variables and classes are loaded
- [x] other top-level child objects trigger errors
- [x] variable expressions cause errors
- [x] classes only support any:: as expression
- [x] default namespace for variables and classes from CMDB data is data, default bundle for variables is variables
- [x] vars, variables can specify namespace and budle and classes can specify namespace
- [x] tags and comment are properly applied from variables and classes
- [x] variables support the direct value specification compatible with vars except for data variables
- [x] variables and classes from CMDB data cannot be re-defined in augments

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6838/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6838/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6838/

**Comment by vpodzime** [comment]
> Sure, I triggered a build:
> 
> [Build Status](https://ci.cfengine.com//job/pr-pipeline/6838/)
> 
> **Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6838/

Only the usual Ubuntu 20 failure.

**Review by larsewi — APPROVED** [review]
Cool tests ;)

File: tests/acceptance/00_basics/06_host_specific_data/01-vars.cf.sub [hunk 1]
```diff
@@ -0,0 +1,6 @@
+bundle agent __main__ {
+  reports:
+      "$(data:variables.test_variable)";
+      "$(with)" with => format("%S", "data:scope1.test_variable");
```
Comment (craigcomstock) [inline tests/acceptance/00_basics/06_host_specific_data/01-vars.cf.sub:4]: cool. I forgot about `format`, I've been using `storejson` a lot recently with `with` for reports. :+1:

File: tests/acceptance/00_basics/06_host_specific_data/10-variable-references.cf.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "variables" : {
+        "test_variable": {
+            "value" : "$(sys.uptime)",
+            "comment" : "comment1",
+            "tags" : ["test_tag"]
+        }
+        "test_variable2": {
```
Comment (craigcomstock) [inline tests/acceptance/00_basics/06_host_specific_data/10-variable-references.cf.json:8]: github clued me in to a missing comma between the variables with a red `:`, invalid JSON but parseable by our parser eh?

**Review by craigcomstock — APPROVED** [review]
looks good! One missing comma in JSON which apparently doesn't break our parser. :p

File: tests/acceptance/00_basics/06_host_specific_data/10-variable-references.cf.json (hunk 2 above)
Comment (vpodzime) [reply tests/acceptance/00_basics/06_host_specific_data/10-variable-references.cf.json:8]: Good catch! There was also one extra comma in the *02-variables.cf.json* file.

**Review by larsewi — APPROVED** [review]
ACK

**Pull request merged**